### PR TITLE
feat(askvm): add useFor resource (lambda function) 

### LINKED ISSUE #43: Handle providing a lambda function when chaining methods in AskScript
```
ask {
	const double = fun(s:string):string {
    s:concat(s)
	}

  text :call(fun(s:string): string {
    s:double // same as s:double()
  })
}
```
returns the value of `text` variable instead of `text` + `text`.

### PR DESCRIPTION
**Pull request by czerwinskilukasz1**
Closes #43

### CODE REVIEW

File: src/askscript/__tests__/03-lambda/program07c-custom_lambda.ask [hunk 1]
```diff
@@ -0,0 +1,9 @@
+ask {
+	let customUseFor = fun(x, f) { f(x) }
```
Comment (mhagmajer) [inline src/askscript/__tests__/03-lambda/program07c-custom_lambda.ask:2]: :+1:

File: src/askjsx/lib/Call.tsx [hunk 2]
```diff
@@ -15,6 +15,21 @@ export function Call({
 }) {
   assert(isString(name), 'name');
   // assert(isStringArray(args), 'args');
+  if (name === 'useFor') {
```
Comment (mhagmajer) [inline src/askjsx/lib/Call.tsx:18]: Please add a comment explaining that this is a temporary way of linking some AskScript code. We'll have to make typing work before introducing a better way

File: src/askjsx/lib/Call.tsx [hunk 3]
```diff
@@ -15,6 +15,24 @@ export function Call({
...
+
+  // This is a temporary way of linking some AskScript code.
+  // We'll have to make typing work before introducing a better way
+  if (name === 'useFor') {
+    assert('params' in args, 'No params found in arguments');
+    const params = (args as any).params;
+    assert(Array.isArray(params), 'params should be an array');
+    assert(params.length == 2, `expecting exactly 2 params for ${name}`);
+
+    if ((args as any).params)
```
Comment (mhagmajer) [inline src/askjsx/lib/Call.tsx:27]: this check is not needed

File: src/askjsx/lib/Call.tsx [hunk 4]
```diff
@@ -15,6 +15,24 @@ export function Call({
...
+      return (
+        <code call>
+          <Ref name={'call'} />
+          {[(args as any).params[1], (args as any).params[0]]}
```
Comment (mhagmajer) [inline src/askjsx/lib/Call.tsx:31]: why not just use `params` here?

Comment (czerwinskilukasz1) [reply src/askjsx/lib/Call.tsx]: Right

**Pull request merged**